File: src/ProjectR.Scripting/content/scripts/spells/SurgeOfLightning.lua
function GetTargetType()
    return Single
end

function GetName()
  return "Surge of Lightning"
end

function GetDescription()
    return "Infuse your target with the power of lightning.\nIncreases AD & MD greatly.\nInflicts PAR."
end

function IsSupportSpell()
    return true
end

function GetMPCost()
    return .4
end

function GetDelay()
  return .6
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
  d:BuffStat(AD, .72)
  d:BuffStat(MD, .72)
  d:ApplyDebuff(PAR, 300)
end

File: src/ProjectR.Scripting/content/scripts/spells/SurgeOfLightning.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class SurgeOfLightning : SpellScriptBase
    {
        public override string Name { get { return "Surge of Lightning"; } }
        public override string Description { get { return "Infuse your target with the power of lightning.\nIncreases AD & MD greatly.\nInflicts PAR."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return true; } }

        public override double MPCost { get { return 40; } }
        public override double Delay { get { return 60; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            BuffStat(Stat.AD, 2);
            BuffStat(Stat.MD, 2);
            TryToApplyDebuff(DebuffResistance.PAR, 300);
        }
    }
}